x-pack/platform/packages/shared/kbn-elastic-assistant/impl/assistant/settings/product_documentation/index.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiButton,
+  EuiCallOut,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLoadingSpinner,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import React, { useEffect, useState, useCallback, useMemo } from 'react';
+import { useInstallProductDoc } from '../../api/product_docs/use_install_product_doc';
+import { useGetProductDocStatus } from '../../api/product_docs/use_get_product_doc_status';
+import * as i18n from './translations';
+
+export const ProductDocumentationManagement: React.FC = React.memo(() => {
+  const [isInstalled, setInstalled] = useState<boolean>(true);
+  const [isInstalling, setInstalling] = useState<boolean>(false);
```

Comment:
Are the new states `isInstalled` and `isInstalling` necessary? I believe they could be removed with something like this which would prevent a re-render: 

```suggestion
  const { mutateAsync: installProductDoc, isLoading: isInstalling } = useInstallProductDoc();
  const { status, isLoading: isStatusLoading } = useGetProductDocStatus();

  const isInstalledOrUnknown = status == undefined ? true : status.overall === 'installed'
  
  ...
  
  if (isInstalledOrUnknown) {
    return null;
  }
```

Reply:
This does not track if `isInstalling`, we need to show loading indicator

Reply:
I went ahead and combined state values to avoid separate state updates triggering multiple renders.